modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5005,7 +5063,7 @@ bool GDScriptAnalyzer::get_function_signature(GDScriptParser::Node *p_source, bo
 	while (base_script.is_valid() && base_script->has_method(function_name)) {
 		MethodInfo info = base_script->get_method_info(function_name);
 
-		if (!(info == MethodInfo())) {
+		if (!info.name.is_empty()) {
```

Comment:
@vreon @AThousandShips `info == MethodInfo()` is always `true` because internally, it compares the `id` field which is rarely changed from the default `0`. E.g. `MethodInfo::from_dict` which is used by `get_method_info` here never sets the `MethodInfo` `id`. 

I don't know what the preferred way of fixing this is here.

Reply:
I would prefer `if (info != MethodInfo())`, since it's clearer. `info.name.is_empty()` is not the most obvious for checking an empty MethodInfo.

Reply:
I have now changed the equality operator of `MethodInfo` to also include the `name` field. I think this matches the implementation of the less-than operator.